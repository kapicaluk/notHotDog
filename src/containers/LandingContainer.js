import React, { Component } from 'react';
import {
    Image,
    StyleSheet,
    Text,
    View,
    Button
} from 'react-native';
import {Actions} from 'react-native-router-flux'

class LandingContainer extends Component {
    constructor(props){
        super(props)
    }
    render() {
        return <View>
            <Button onPress={() => this.props.signOutUser() } title="SignOut"/>
            <Button onPress={() => {Actions.tabbar()}} title="Account" />
            <Button onPress={() => {Actions.gallery()}} title="Camera" />
          </View>;
    }
}

export default LandingContainer
