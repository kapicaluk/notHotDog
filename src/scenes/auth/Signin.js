import React, { Component } from "react";
import PropTypes from "prop-types";
import { Text, View, Image, KeyboardAvoidingView } from "react-native";
import { connect } from "react-redux";
import { signInUser, clearState } from "../../store/auth";
import { Actions } from "react-native-router-flux";
import { Field, reduxForm } from "redux-form";
import { Container, Input, Button, Item, Spinner } from "../../components/common/index";
import logo from '../../../logo.png';
import styles from "./authStyle";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  clearState: PropTypes.func.isRequired,
  signInUser: PropTypes.func.isRequired,
  authError: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

const background = require("../../../starwberry2.jpg");

class Signin extends Component {
  constructor(props) {
    super(props);

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  componentWillMount() {
    if (!this.props.userId) this.props.clearState();
  }

  handleFormSubmit(props) {
    const { email, password } = props;

    this.props.signInUser({ email, password });
  }

  render() {
    const { handleSubmit } = this.props;

    return (
      <Image source={background} style={styles.background} resizeMode="cover">
        <KeyboardAvoidingView>
        <View style={styles.container}>
          <Container>
            {/*<Image source={logo} style={styles.logoStyle} />*/}
            <Item>
              <Field
                name="email"
                component={Input}
                placeholder="Email"
                autoCapitalize={"none"}
              />
            </Item>
          </Container>
          <Container>
            <Item>
              <Field
                name="password"
                component={Input}
                secureTextEntry
                placeholder="Password"
              />
            </Item>
          </Container>
          {this.props.authError ? (
            <Text style={styles.error}>{this.props.authError}</Text>
          ) : (
            <View />
          )}
          <Container>
            {this.props.loading ? (
              <Item style={styles.loadingContainer}>
                <Spinner />
              </Item>
            ) : (


              <Button onPress={handleSubmit(this.handleFormSubmit)}>
                Log in
              </Button>


            )}

            <Button onPress={() => Actions.signup()}>
              Sign Up
            </Button>
          </Container>
        </View>
        </KeyboardAvoidingView>
      </Image>
    );
  }
}

const validate = props => {
  const errors = {};
  const fields = ["email", "password"];

  fields.forEach(f => {
    if (!(f in props)) {
      errors[f] = `${f} is required`;
    }
  });

  return errors;
};

Signin.propTypes = propTypes;
Signin = reduxForm({ form: "signin", validate })(Signin);

const mapStateToProps = ({ auth }) => ({
  authError: auth.error,
  loading: auth.loading,
  userId: auth && auth.user ? auth.user.uid : ""
});

export default connect(mapStateToProps, { signInUser, clearState })(Signin);
