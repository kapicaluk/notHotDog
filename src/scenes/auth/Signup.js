import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Text, View, TouchableOpacity, Image } from "react-native";
import { Actions } from "react-native-router-flux";
import { Field, reduxForm } from "redux-form";
import {
  Container,
  Input,
  Button,
  Item,
  Spinner
} from "../../components/common/index";
import styles from "./authStyle";
import { signUpUser, clearState } from "../../store/auth";

const propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  clearState: PropTypes.func.isRequired,
  signUpUser: PropTypes.func.isRequired,
  authError: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

const background = require("../../../starwberry2.jpg");

class Signup extends Component {
  constructor(props) {
    super(props);

    this.handleFormSubmit = this.handleFormSubmit.bind(this);
  }

  componentWillMount() {
    this.props.clearState();
  }

  handleFormSubmit(props) {
    const { email, password, firstname, lastname } = props;

    this.props.signUpUser({ email, password, firstname, lastname });
  }

  render() {
    const { handleSubmit } = this.props;

    return <Image source={background} style={styles.background} resizeMode="cover">
        <View style={styles.container}>
          <Container>
            <Item>
              <Field name="firstname" component={Input} placeholder="First name" />
            </Item>

            <Item>
              <Field name="lastname" component={Input} placeholder="Last name" />
            </Item>

            <Item>
              <Field name="email" component={Input} placeholder="Email" autoCapitalize={"none"} />
            </Item>

            <Item>
              <Field name="password" component={Input} secureTextEntry placeholder="Password" />
            </Item>

            <Item>
              <Field name="repassword" component={Input} secureTextEntry placeholder="Repeat Password" />
            </Item>
            {this.props.authError ? <Text style={styles.error}>
                {this.props.authError}
              </Text> : <View />}

            {this.props.loading ? <Item style={styles.loadingContainer}>
                <Spinner />
              </Item> : <Container>
                <Button onPress={() => Actions.signup()}>Sign Up</Button>
              </Container>}
          </Container>
        </View>
      </Image>;
  }
}

const validate = props => {
  const errors = {};
  const fields = ["firstname", "lastname", "email", "password"];

  fields.forEach(f => {
    if (!(f in props)) {
      errors[f] = `${f} is required`;
    }
  });

  if (props.firstname && props.firstname.length < 3) {
    errors.firstname = "Minimum of 3 characters";
  } else if (props.firstname && props.firstname.length > 20) {
    errors.firstname = "Maximum of 20 characters";
  }

  if (props.lastname && props.lastname.length < 3) {
    errors.lastname = "Minimum of 3 characters";
  } else if (props.lastname && props.lastname.length > 20) {
    errors.lastname = "Maximum of 20 characters";
  }

  if (
    props.email &&
    !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(props.email)
  ) {
    errors.email = "please provide valid email";
  }

  if (props.password && props.password.length < 6) {
    errors.password = "minimum 6 characters";
  }

  if (props.password !== props.repassword) {
    errors.repassword = "passwords doesn't match";
  }

  return errors;
};

Signup.propTypes = propTypes;
Signup = reduxForm({ form: "signup", validate })(Signup);

const mapStateToProps = ({ auth }) => {
  const { error, loading, user } = auth;

  return { authError: error, loading, user };
};

export default connect(mapStateToProps, { signUpUser, clearState })(Signup);
