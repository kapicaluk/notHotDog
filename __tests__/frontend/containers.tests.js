import React from 'react';
import requireAuth from '../../src/containers/auth/requireAuth';
import requireNotAuth from '../../src/containers/auth/requireNotAuth';
import Signup from '../../src/containers/auth/Signup';
import Signin from '../../src/containers/auth/Signin';
import LandingContainer from '../../src/containers/LandingContainer';

import renderer from 'react-test-renderer';

/* Auth Containers */
describe('requireAuth', () => {
  it('renders without crashing', () => {
    const renderedRequireAuth = renderer.create(<requireAuth />).toJSON();
    expect(renderedRequireAuth).toMatchSnapshot();
  })
})

describe('requireNotAuth', () => {
  it('renders without crashing', () => {
    const renderedRequireNotAuth = renderer.create(<requireNotAuth />).toJSON();
    expect(renderedRequireNotAuth).toMatchSnapshot();
  })

  it('renders signup component', () => {
    const renderedSignup = requireNotAuth(Signup);
    expect(renderedSignup).toMatchSnapshot();
  })

  it('renders signin component', () => {
    const renderedSignin = requireNotAuth(Signin);
    expect(renderedSignin).toMatchSnapshot();
  })
})

/* Landing Container */
describe('LandingContainer', () => {
  it('renders without crashing', () => {
    const renderedLandingContainer = renderer.create(<LandingContainer />).toJSON();
    expect(renderedLandingContainer).toMatchSnapshot();
  });
})