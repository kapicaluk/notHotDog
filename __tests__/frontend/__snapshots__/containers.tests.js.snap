// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingContainer renders without crashing 1`] = `
<View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        SignOut
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Account
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Camera
      </Text>
    </View>
  </View>
</View>
`;

exports[`requireAuth renders without crashing 1`] = `<requireAuth />`;

exports[`requireNotAuth renders signin component 1`] = `[Function]`;

exports[`requireNotAuth renders signup component 1`] = `[Function]`;

exports[`requireNotAuth renders without crashing 1`] = `<requireNotAuth />`;
